File: Extensions/Tessa.Extensions.Default.Client/UI/CardFiles/FilesViewCardControlInitializationStrategy.cs
using System.Threading.Tasks;
using Tessa.Properties.Resharper;
using Tessa.UI.Cards.Controls;
using Tessa.UI.Menu;
using Tessa.Views;

namespace Tessa.Extensions.Default.Client.UI.CardFiles
{
    /// <summary>
    /// Стратегия инициализации модели представления
    /// </summary>
    public class FilesViewCardControlInitializationStrategy : ViewCardControlInitializationStrategy
    {

        #region Constructor

        public FilesViewCardControlInitializationStrategy([NotNull] IViewService viewService,
            [NotNull] CreateMenuContextFunc createMenuContextFunc,
            [NotNull] IViewCardControlContentItemsFactory contentItemsFactory)
            : base(viewService, createMenuContextFunc, contentItemsFactory)
        { }

        #endregion

        #region Base overrides
        
        public override ValueTask InitializeMetadataAsync(CardViewControlInitializationContext context)
        {
            context.ControlViewModel.ViewMetadata = FilesViewMetadata.Create();

            return new ValueTask();
        }
        public override ValueTask InitializeDataProviderAsync(CardViewControlInitializationContext context)
        {
            context.ControlViewModel.DataProvider =
                new CardFilesDataProvider(
                    context.ControlViewModel.ViewMetadata,
                    FilesViewGeneratorBaseUIExtension.TryGetFileControl(context.Model.Info, context.ControlViewModel.Name));

            return new ValueTask();
        }

        #endregion
    }
}

File: WebClient SDK/src/default/ui/cardFiles/filesViewCardControlInitializationStrategy.ts
import { FilesViewMetadata } from './filesViewMetadata';
import { CardFilesDataProvider } from './cardFilesDataProvider';
import {
  ViewControlInitializationStrategy,
  ViewControlInitializationContext,
  FileListViewModel
} from 'tessa/ui/cards/controls';
import { tryGetFromInfo } from 'tessa/ui';

export class FilesViewCardControlInitializationStrategy extends ViewControlInitializationStrategy {
  initializeMetadata(context: ViewControlInitializationContext) {
    context.controlViewModel.viewMetadata = FilesViewMetadata.create();
  }

  initializeDataProvider(context: ViewControlInitializationContext) {
    const fileControl = tryGetFromInfo<FileListViewModel | null>(
      context.model.info,
      context.controlViewModel.name ?? '',
      null
    );
    if (fileControl) {
      context.controlViewModel.dataProvider = new CardFilesDataProvider(
        context.controlViewModel.viewMetadata!,
        fileControl
      );
    }
  }
}
